pkg/model/bootstrapscript.go

```diff
@@ -158,7 +184,84 @@ func (b *BootstrapScript) getRelevantHooks(hooks []kops.HookSpec, role kops.Inst
 			}
 		}
 	}
-	return relevantHooks
+
+	hooks := []kops.HookSpec{}
+	if len(relevantHooks) > 0 {
+		for _, hook := range relevantHooks {
+			if hook.Manifest != "" {
+				manifestFingerprint, err := b.computeFingerprint(hook.Manifest)
+				if err != nil {
+					return nil, err
+				}
+				hook.Manifest = manifestFingerprint + " (fingerprint)"
+			}
+
+			if hook.ExecContainer != nil && hook.ExecContainer.Command != nil {
+				execContainerCommandFingerprint, err := b.computeFingerprint(hook.ExecContainer.Command)
+				if err != nil {
+					return nil, err
+				}
+				hook.ExecContainer.Command = []string{execContainerCommandFingerprint + " (fingerprint)"}
+			}
+
+			hook.Roles = nil
+			hooks = append(hooks, hook)
+		}
+	}
+
+	return hooks, nil
+}
+
+// getRelevantFileAssets returns a list of file assets to be applied to the
+// instance group, with the Content fingerprinted to reduce size
+func (b *BootstrapScript) getRelevantFileAssets(allFileAssets []kops.FileAssetSpec, role kops.InstanceGroupRole) ([]kops.FileAssetSpec, error) {
+	relevantFileAssets := []kops.FileAssetSpec{}
+	for _, fileAsset := range allFileAssets {
+		if len(fileAsset.Roles) == 0 {
+			relevantFileAssets = append(relevantFileAssets, fileAsset)
+			continue
+		}
+		for _, fileAssetRole := range fileAsset.Roles {
+			if role == fileAssetRole {
+				relevantFileAssets = append(relevantFileAssets, fileAsset)
+				break
+			}
+		}
+	}
+
+	fileAssets := []kops.FileAssetSpec{}
+	if len(relevantFileAssets) > 0 {
+		for _, fileAsset := range relevantFileAssets {
+			if fileAsset.Content != "" {
+				contentFingerprint, err := b.computeFingerprint(fileAsset.Content)
+				if err != nil {
+					return nil, err
+				}
+				fileAsset.Content = contentFingerprint + " (fingerprint)"
```

Comment:
Long term I feel like it would be safer to use a hash in a new field (`ContentHash`), but this is the right way to start!

Reply:
Sure that would be better. Would you suggest updating the spec to include ContentHash (which may get confusing, as we won't expect users to be specifying this in their spec), or just add it as a new field here (might be a bit complicated as it's not a valid field on the struct)?